presto-hive/src/test/java/com/facebook/presto/hive/TestOrcFileWriterConfig.java

```diff
@@ -65,4 +76,71 @@ public void testExplicitPropertyMappings()
 
         assertFullMapping(properties, expected);
     }
+
+    static class SetAndVerifyConfig
+            implements Callable<Boolean>
+    {
+        private final OrcFileWriterConfig config;
+
+        SetAndVerifyConfig(OrcFileWriterConfig config)
+        {
+            this.config = config;
+        }
+
+        @Override
+        public Boolean call()
+        {
+            for (int i = 0; i < 100; i++) {
+                OrcWriterOptions.Builder builder = config.toOrcWriterOptionsBuilder();
+                int value = ThreadLocalRandom.current().nextInt(32);
+                DataSize stripeMinSize = new DataSize(10 + value, MEGABYTE);
+                DataSize stripeMaxSize = new DataSize(50 + value, MEGABYTE);
+                DataSize dictionaryMaxMemory = new DataSize(20 + value, MEGABYTE);
+                int stripeMaxRowCount = 1_000_000 + value;
+                DataSize stringStatisticsLimit = new DataSize(32 + value, BYTE);
+
+                builder.withStripeMinSize(stripeMinSize)
+                        .withStripeMaxSize(stripeMaxSize)
+                        .withStripeMaxRowCount(stripeMaxRowCount)
+                        .withDictionaryMaxMemory(dictionaryMaxMemory)
+                        .withMaxStringStatisticsLimit(stringStatisticsLimit);
+
+                sleepUninterruptibly(value % 10, MILLISECONDS);
+
+                OrcWriterOptions options = builder.build();
+                assertEquals(stripeMinSize, options.getStripeMinSize());
+                assertEquals(stripeMaxSize, options.getStripeMaxSize());
+                assertEquals(stripeMaxRowCount, options.getStripeMaxRowCount());
+                assertEquals(dictionaryMaxMemory, options.getDictionaryMaxMemory());
+                assertEquals(stringStatisticsLimit, options.getMaxStringStatisticsLimit());
+            }
+            return true;
+        }
+    }
+
+    @Test
+    public void testConcurrentOptions()
+            throws Exception
+    {
+        int totalThreads = 10;
+        OrcFileWriterConfig config = new OrcFileWriterConfig()
+                .setStripeMinSize(new DataSize(13, MEGABYTE))
+                .setStripeMaxSize(new DataSize(27, MEGABYTE))
+                .setStripeMaxRowCount(44)
+                .setRowGroupMaxRowCount(11)
+                .setDictionaryMaxMemory(new DataSize(13, MEGABYTE))
+                .setStringStatisticsLimit(new DataSize(17, MEGABYTE))
+                .setMaxCompressionBufferSize(new DataSize(19, MEGABYTE));
+
+        ImmutableList.Builder<SetAndVerifyConfig> builder = ImmutableList.builder();
+        for (int i = 0; i < totalThreads; i++) {
+            builder.add(new SetAndVerifyConfig(config));
+        }
+        ExecutorService executorService = newFixedThreadPool(totalThreads);
+        for (Future<Boolean> result : executorService.invokeAll(builder.build())) {
+            result.get();
```

Comment:
This will cause waiting on the future, so it will be run in a single thread. But honestly it feels like this test is an overkill. Multithreaded testing is hard and often flaky and error prone. If it is possible to avoid it - better to avoid it. But if you feel strongly about this test - feel free to keep it. Just make sure it does what you intended it to do.

Reply:
This is my understanding, The threads will run in parallel, but the main thread will wait for them sequentially. So it would produce errors if there are conflicts and I verified before writing the test that it indeed produced conflict. But with the move to the config maintaining its own state, this is not required and I have removed this test.